src/modules/esl-utils/dom/scroll/parent.ts

```diff
@@ -3,32 +3,36 @@ import {isElement, getNodeName, getParentNode} from '../api';
 /**
  * Get the list of all scroll parents, up the list of ancestors until we get to the top window object.
  * @param element - element for which you want to get the list of all scroll parents
- * @param list - array of elements to concatenate with the list of all scroll parents of element (optional)
+ * @param topElement - an element which element considered a final scrollable parent target (optional, defaults to element.ownerDocument?.body)
  */
-export function getListScrollParents(element: Element, list: Element[] = []): Element[] {
-  const scrollParent = getScrollParent(element);
-  const isBody = scrollParent === element.ownerDocument?.body;
-  const target = isBody
-    ? isScrollable(scrollParent) ? scrollParent : []
-    : scrollParent;
-
-  const updatedList = list.concat(target);
-  return isBody
-    ? updatedList
-    : updatedList.concat(getListScrollParents(getParentNode(scrollParent) as Element));
+export function getListScrollParents(element: Element, topElement?: Element): Element[] {
+  const actualTarget = topElement || element.ownerDocument?.body;
+  const scrollParent = getScrollParent(element, actualTarget);
+  if (!scrollParent) return [];
+  const isScrollableTarget = scrollParent === actualTarget;
+  if (isScrollableTarget) return isScrollable(scrollParent) ? [scrollParent] : [];
+  return [scrollParent].concat(getListScrollParents(getParentNode(scrollParent) as Element, actualTarget));
 }
 
+/**
+ * Get the scroll parent of the specified element in the DOM tree.
+ * @param node - element for which to get the scroll parent
+ * @param topElement - element for which to get the scroll parent
+ */
+export function getScrollParent(node: Element, topElement: Element): Element | undefined;
 /**
  * Get the scroll parent of the specified element in the DOM tree.
  * @param node - element for which to get the scroll parent
  */
-export function getScrollParent(node: Element): Element {
+export function getScrollParent(node: Element): Element;
+export function getScrollParent(node: Element, target?: Element): Element | undefined {
```

Comment:
```suggestion
export function getScrollParent(node: Element, topElement?: Element): Element | undefined {
```
or
```suggestion
export function getScrollParent(node: Element, boundary?: Element): Element | undefined {
```

It is the internal variable, but nevertheless, it should be called meaningfully